### PR TITLE
Add deprecation warning to ModelCheckpoint when logging val_loss with no monitor

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Adds missing deprecation warning.

Context: https://github.com/PyTorchLightning/pytorch-lightning/pull/6004#issuecomment-779810628

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR

### CODE REVIEW

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 1]
```diff
@@ -185,9 +188,6 @@ def __init__(
         self.save_function = None
         self.warned_result_obj = False
 
-        if save_top_k is None and monitor is not None:
-            self.save_top_k = 1
```
Comment (carmocca) [inline pytorch_lightning/callbacks/model_checkpoint.py:189]: This is already done in `_add_backward_monitor_support`

**Review by Borda — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

File: tests/deprecated_api/test_remove_1-4.py [hunk 2]
```diff
@@ -189,3 +189,18 @@ def test_v1_4_0_deprecated_lightning_data_parallel():
...
+def test_v1_4_0_deprecated_checkpoint_on(tmpdir):
+    from pytorch_lightning.callbacks.model_checkpoint import warning_cache
+    warning_cache.clear()
+
+    class TestModel(BoringModel):
+        def training_step(self, batch, batch_idx):
+            self.log("val_loss", -batch_idx)
+            return super().training_step(batch, batch_idx)
+
+    trainer = Trainer(default_root_dir=tmpdir, checkpoint_callback=True, max_epochs=1)
```
Comment (rohitgr7) [inline tests/deprecated_api/test_remove_1-4.py:203]:
```suggestion
    trainer = Trainer(default_root_dir=tmpdir, callbacks=[ModelCheckpoint()], max_epochs=1)
```
checkpoint_callback will be removed in 1.3. Also, you might need to import `ModelCheckpoint` here.

Comment (carmocca) [reply tests/deprecated_api/test_remove_1-4.py:203]: `checkpoint_callback` won't be removed, what will be removed is passing an instance of `ModelCheckpoint` to `checkpoint_callback`

Comment (rohitgr7) [reply tests/deprecated_api/test_remove_1-4.py:203]: correct :+1:

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**